src/main/java/io/kontur/disasterninja/controller/AppsController.java

```diff
@@ -126,14 +127,20 @@ public List<LayerDetailsDto> updateListOfLayers(@PathVariable("id") UUID appId,
         return updatedLayers.stream().map(LayerDetailsDto::fromLayer).toList();
     }
 
-    @Operation(summary = "Get application config with features and user settings by id. Returns default app if no " +
-            "appId is provided and requester domain is unknown", tags = {"Applications"})
+    @Operation(summary = "Get application config with features and user settings by id. Returns default app if " +
+            "no appId is provided and requester domain is unknown", tags = {"Applications"})
     @ApiResponse(responseCode = "200",
             content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
                     schema = @Schema(implementation = AppDto.class)))
     @GetMapping(path = "/configuration")
     public AppDto getAppConfig(@RequestParam(name = "appId", required = false) UUID appId,
-                               @RequestHeader(name = "Host", required = false) String domain) {
+                               @RequestHeader(name = "referer", required = false) String referer) {
```

Comment:
Referer is not hostname that your endpoint is at but rather the one that issues tge request, and can be skipped by the browser in certain conditions. Tge correct header is Host really. Why did you change to referer?

Reply:
Because `Host` header contains inner k8s link instead of the initial hostname. https://konturio.slack.com/archives/C037ZB1Q53P/p1683712911568459?thread_ts=1683712631.715549&cid=C037ZB1Q53P
If not using `Referer` header, the option that I see is for FE to pass the desired value in additional parameter or in another header.

Reply:
Okay, if you have proxy chains that can't set Host properly (why? nginx can) then at least use X-Forwarded-Host https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host

Reply:
Sure. Additional changes are required then, probably in nginx config.